wrapper/virt_v2v_wrapper.py

```diff
@@ -347,11 +352,26 @@ def main():
                 data, host.get_uid(), host.get_gid())
             if agent_pid is None:
                 raise RuntimeError('Failed to start ssh-agent')
+        source_host = detect_source_host(data, agent_sock)
+        if source_host:
+            try:
+                source_host.prepare_exports()
+            except Exception as e:
+                error_name = e.args[0] if e.args else "Wrapper failure"
+                error(error_name, 'Failed to prepare NBD exports!',
+                      exception=True)
+                source_host.close_exports()
+                STATE.failed = True
         if STATE.pre_copy:
             host.prepare_disks(data)
             STATE.pre_copy.copy_disks(data['vmware_password_file'])
         if not STATE.failed:
-            wrapper(host, data, virt_v2v_caps, agent_sock)
+            if source_host and source_host.avoid_wrapper(host):
+                source_host.transfer_exports(host)
```

Comment:
... even though it is basically the same as this.  Seems like a lot of duplication and extra, unclean, hard-to-read branching.